code-studio/build-commands.js

```diff
@@ -49,10 +49,14 @@ exports.browserify = function (config) {
   var command = (shouldWatch ? 'watchify -v' : 'browserify') +
     (shouldMinify ? '' : ' --debug');
 
+  var reactifyStep = '-t reactify';
+
+  var globifyStep = '-t require-globify';
+
   var factorStep = '';
   if (shouldFactor) {
     factorStep = "-p [ factor-bundle -o '" + (shouldMinify ? 'uglifyjs ' : '') +
-      '>' + buildPath + "`basename $FILE .js`" + extension + "']";
+      '>' + buildPath + "`dirname $(realpath --relative-to " + srcPath + " $FILE)`/`basename -s .js -s .jsx $FILE`" + extension + "']";
```

Comment:
Changed this to work better and commented it much better.  See the code for full description; in short, this is the output filename for any given input file:

```bash
./build/js/`realpath --relative-to ./src/js/ $FILE | sed -r "s/.jsx?$/.js/i"`
```